cmd/internal/encryption/encryption.go

```diff
@@ -0,0 +1,241 @@
+		return "", err
+	}
+
+	if err := e.decryptFile(infile, outfile, block, iv, msgLen); err != nil {
+		return "", err
+	}
+
+	if err := e.fs.Remove(inputPath); err != nil {
+		e.log.Warn("unable to remove input", "error", err)
+	}
+	return output, nil
+}
+
+func isASCII(s string) bool {
+	for _, c := range s {
+		if c > unicode.MaxASCII {
+			return false
+		}
+	}
+	return true
+}
+
+// createCipher() returns new cipher block for encryption/decryption based on encryption-key
+func (e *Encrypter) createCipher() (cipher.Block, error) {
+	key := []byte(e.key)
+	return aes.NewCipher(key)
+}
+
+func (e *Encrypter) openOutputFile(output string) (afero.File, error) {
+	return e.fs.OpenFile(output, os.O_RDWR|os.O_CREATE, 0644)
+}
+
+// generateIV() returns unique initalization vector of same size as cipher block for encryption
+func (e *Encrypter) generateIV(block cipher.Block) ([]byte, error) {
+	iv := make([]byte, block.BlockSize())
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		return nil, err
+	}
+	return iv, nil
+}
+
+// encryptFile() encrypts infile to outfile using CTR mode (cipher and iv) and appends iv for decryption
+func (e *Encrypter) encryptFile(infile, outfile afero.File, block cipher.Block, iv []byte) error {
+	buf := make([]byte, 1024)
+	stream := cipher.NewCTR(block, iv)
+
+	for {
+		n, err := infile.Read(buf)
+		if n > 0 {
+			stream.XORKeyStream(buf, buf[:n])
+			if _, err := outfile.Write(buf[:n]); err != nil {
+				return err
+			}
+		}
+
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			e.log.Info("read %d bytes: %s", strconv.Itoa(n), err)
+			break
+		}
+	}
+
+	if _, err := outfile.Write(iv); err != nil {
+		return fmt.Errorf("could not append iv: %w", err)
+	}
+
+	return nil
+}
+
+// IsEncrypted() tests if target file is encrypted
+func IsEncrypted(path string) bool {
+	return filepath.Ext(path) == suffix
+}
+
+// readIVAndMessageLength() returns initialization vector and message length for decryption
+func (e *Encrypter) readIVAndMessageLength(infile afero.File, block cipher.Block) ([]byte, int64, error) {
+	fi, err := infile.Stat()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	iv := make([]byte, block.BlockSize())
+	msgLen := fi.Size() - int64(len(iv))
+	if _, err := infile.ReadAt(iv, msgLen); err != nil {
+		return nil, 0, err
+	}
+
+	return iv, msgLen, nil
+}
+
+// decryptFile() decrypts infile to outfile using CTR mode (cipher and iv)
+func (e *Encrypter) decryptFile(infile, outfile afero.File, block cipher.Block, iv []byte, msgLen int64) error {
+	buf := make([]byte, 1024)
+	stream := cipher.NewCTR(block, iv)
+
+	for {
+		n, err := infile.Read(buf)
+		if n > 0 {
+			if n > int(msgLen) {
+				n = int(msgLen)
+			}
+			msgLen -= int64(n)
+			stream.XORKeyStream(buf, buf[:n])
+			if _, err := outfile.Write(buf[:n]); err != nil {
+				return err
+			}
+		}
+
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			e.log.Info("read %d bytes: %s", strconv.Itoa(n), err)
+			break
```

Comment:
Maybe it would be more secure to do this?

```suggestion
			return fmt.Errorf("error reading from file (%d bytes read): %w", strconv.Itoa(n), err)
```